internal/service/elbv2/target_group.go

```diff
@@ -263,6 +265,31 @@ func ResourceTargetGroup() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validation.StringInSlice(elbv2.TargetGroupIpAddressTypeEnum_Values(), false),
 			},
+			"target_failover": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"on_deregistration": {
+							Type:     schema.TypeString,
+							Optional: true,
```

Comment:
Since these parameters must both be present if `target_failover` is defined, they should be marked as `Required`